odoo/addons/base/models/ir_module.py

```diff
@@ -919,6 +919,19 @@ def _check(self):
             if not module.description_html:
                 _logger.warning('module %s: description is empty !', module.name)
 
+    def _get(self, name):
+        """ Return the (sudoed) `ir.module.module` record with the given name.
+        The result may be an empty recordset if the module is not found.
+        """
+        model_id = self._get_id(name) if name else False
+        return self.sudo().browse(model_id)
+
+    @tools.ormcache('name')
+    def _get_id(self, name):
+        self.env.cr.execute("SELECT id FROM ir_module_module WHERE name=%s", (name,))
+        result = self.env.cr.fetchone()
+        return result and result[0]
```

Comment:
This can just return the `fetchone()` directly, it should be either `None` (which is going to browse to an empty recordset) or a 1-uple (which is going to browse to a singleton).

Reply:
I was following the `ir.model._get` & `_get_id` definition, but I'll use an improved version then :D.